<?xml version="1.0" encoding="UTF-8"?>
<launch>

  
 
  
  <arg name="model_path" value="/home/vijay/thesis/repos/scnet_r50_fpn_1x_coco-c3f09857.pth"/>
  <arg name="cfg_path" value="/home/wms/2d_object_detection/mmmdetection/configs/scnet/scnet_r50_fpn_1x_coco.py"/>
  <arg name="label_path" value="/home/wms/2d_object_detection/mmdet_ws/src/mmdet_deploy/data/coco_ids.names"/>
  <arg name="input_image_topic" value="/rgb_cam_01/image_rect_color_sync"/>
  <arg name="threshold_score" value= "0.05"/>
   
 

  <!-- rosbag play -->
  <include file="$(find mmdet_deploy)/launch/play_rosbag.launch" />

  

 

  <node pkg="mmdet_deploy" type="mmdet_node.py" name="mmdet_rgb"  output="screen" required="true">
    <param name="model_path" value="$(arg model_path)"/>
    <param name="cfg_file" value="$(arg cfg_path)"/>
    <param name="label_path" value="$(arg label_path)"/>
    <remap from="input_image_topic" to ="$(arg input_image_topic)"/>
    <param name="threshold_score" value="$(arg threshold_score)"/>
    <param name="detection_name" value="person"/>

    <param name="device" value="cuda:1"/>
    <remap from="detection_image" to="/mmdet_rgb/detection_image"/>
    <remap from="check_for_objects" to="/mmdet_rgb/check_for_objects"/>
    
  </node>

  
  
  

</launch>